Updated submitting patch page: to clarify patches type

mharyam opened the pull request:
# Trac ticket number
Will raise on Django later if this is fine, this is a Draft PR re conversion [here](https://forum.djangoproject.com/t/feedback-needed-updating-documentation-contribution-guidelines/33448/8) 

# Checklist
- [ ] This PR targets the `main` branch. <!-- Backports will be evaluated and done by mergers, when necessary. -->
- [ ] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [ ] I have checked the "Has patch" ticket flag in the Trac system.
- [ ] I have added or updated relevant tests.
- [ ] I have added or updated relevant docs, including release notes if applicable.
- [ ] I have attached screenshots in both light and dark modes for any UI changes.

mharyam committed:
Updated submitting patch page: to clarify patches type

RealOrangeOne commented on docs/internals/contributing/writing-code/submitting-patches.txt:
```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
+
+DEPs are the primary mechanisms for proposing major new features,
+for collecting community input on issues, and for documenting
+design decisions that have gone into Django.
```
> for collecting community input on issues

I'm not sure this is quite right. The forum is a much better place to collect input. A DEP is more about collecting feedback on larger changes. 

```suggestion
DEPs are the primary mechanisms for proposing and collecting community input on 
major new features, and for documenting design decisions that have gone into Django.
```

RealOrangeOne commented on docs/internals/contributing/writing-code/submitting-patches.txt:
```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
+Once the DEP is ready, the :ref:`Steering Council <steering-council>`
+votes on whether to accept it.
+
+You may be encouraged to write a DEP following a discussion on the
```
Suggestion: Perhaps worth highlighting here that DEPs probably shouldn't be the first port of call - instead opening up a forum thread on "Django Internals"? It's sort-of mentioned here, but being explicit should keep the DEP spam down, and allow triage.

mharyam replied:
Thank you, I have updated it. Do you think I should mention why a discussion should be created? This is the line I am referring to  

> This allows the community to provide feedback and helps refine the proposal

RealOrangeOne commented on docs/internals/contributing/writing-code/submitting-patches.txt:
```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
+If you need user feedback on a public interface, landing this into Django core
+is not the way to do it. Iterations must follow the
+:ref:`deprecation process<internal-release-deprecation-policy>`
+and take months-years for the users to see the benefit of such changes.
```
```suggestion
and take months or even years for the users to see the benefit of such changes.
```

mharyam commented:
Thank you @RealOrangeOne for reviewing. I have updated changes and added a small comment above to confirm if I should include a line. If this is fine, I will proceed to create a PR on Django

mharyam commented on docs/internals/contributing/writing-code/submitting-patches.txt:
```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
+Before considering writing a DEP, its recommeneded to first open a discussion
+on the `Django Forum`_ or |django-developers| mailing list.
+
+This allows the community to provide feedback and helps refine the proposal.
```
This is the line  @RealOrangeOne
